components/diagnosis-suggestion/api/src/main/java/org/phenotips/diagnosis/internal/DefaultDiagnosisService.java

```diff
@@ -204,9 +208,14 @@ public int compare(Integer o1, Integer o2)
 
     private void addTermAndAncestors(Term t, Observations o)
     {
-        int id = this.boqa.getTermIndex(t);
-        o.observations[id] = true;
-        this.boqa.activateAncestors(id, o.observations);
+        try {
+            int id = this.boqa.getTermIndex(t);
+            o.observations[id] = true;
+            this.boqa.activateAncestors(id, o.observations);
+        } catch (NullPointerException e) {
+            this.logger.warn(String.format(
+                "Unable to find the boqa index of '%s'.", t));
```

Comment:
`org.slf4j.Logger` has a really nice feature, [_parameterized logging_](http://slf4j.org/faq.html#logging_performance). You should write this as:

    this.logger.warn("Unable to find the boqa index of [{}].", t);

`{}` is an anchor, and as a convention we put them inside square brackets to make it easier to delimit variable values from the fixed logging message.